sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystTypeConverters.scala

```diff
@@ -75,7 +75,7 @@ object CatalystTypeConverters {
       case LongType => LongConverter
       case FloatType => FloatConverter
       case DoubleType => DoubleConverter
-      case DayTimeIntervalType => DurationConverter
+      case _: DayTimeIntervalType => DurationConverter
```

Comment:
let's add a TODO and truncate the Duration if it doesn't match the fields.

Reply:
Here is the JIRA: https://issues.apache.org/jira/browse/SPARK-35726